gaj23/face-it
# Ch/feature/form component

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. 

Add search functionality in search bar Form component to accept tags and brand names.  

### Problems Encountered & Resulting Solutions  
> State any problems encountered 
> Describe your code changes in detail for reviewers.

A lot of refactoring was done in the search method in App to accept the userInput data.  It needs more testing but the main functionality is there for filtering tags and brands for each makeup product without producing duplicate items upon a single search.

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

Form.js, SearchResults.js, and SearchResult.js were all added as new components.  Styling for each of these components was added, included to the Cosmetic and Items components.  App was updated with the search feature and rendering.  

### How should this be tested?
> What steps need to be taken to confirm functionality? 

Run `npm start` in your terminal when the project is open.  Run the server to connect the data.  Type into the search bar and click the button to submit your search.  You should see items show up that match.

### Steps To Take
> Any questions and possible steps moving forward.

Router is not implemented yet to create a unique URL for these search results.  There needs to be some more error handling to take care of multiple search requests or for user handling across pages coming from a search. 

#### Checklist:
